synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -489,13 +492,12 @@ def update_cache(self, sequence, members, rules_by_user, state_group):
             self.state_group = state_group
 
 
-class _Invalidation(namedtuple("_Invalidation", ("cache", "room_id"))):
-    # We rely on _CacheContext implementing __eq__ and __hash__ sensibly,
-    # which namedtuple does for us (i.e. two _CacheContext are the same if
-    # their caches and keys match). This is important in particular to
-    # dedupe when we add callbacks to lru cache nodes, otherwise the number
-    # of callbacks would grow.
```

Comment:
Is this comment no longer relevant?

Reply:
oh, I didn't understand it so removed it. If you know what it means, I can reword?

Reply:
(how is `_CacheContext` involved here?)

Reply:
I think what it is meant to be saying is that we store invalidation callbacks as a set in `LruCache`, so if each time we pass in an invalidation callback that has a different hash then that set of callbacks will grow each time we read from it. 

https://github.com/matrix-org/synapse/blob/0e5fa9cc245249350ef756bee0474bde74ea634b/synapse/util/caches/lrucache.py#L246

I have a suspicion that the mention of `_CacheContext` is because it was cargo culted from the comment in `_CacheContext` as it does a similar thing.

https://github.com/matrix-org/synapse/blob/0e5fa9cc245249350ef756bee0474bde74ea634b/synapse/util/caches/descriptors.py#L406-L408

Reply:
right, thanks. Have added a rewritten comment.